### ISSUE TITLE
WordPress 6.1 Compatibility

### ISSUE THREAD

**Issue by swissspidy** [swissspidy/web-stories-wp#12223]
## Summary

**Before release of WordPress 6.1:**

* [ ] Manual testing against trunk and Beta releases
* [ ] Run E2E tests against 6.1 RC once published
* [ ] Update staging environments to use 6.1 RC
* [ ] Bump Tested up to in readme.txt

**After release of WordPress 6.1:**

* [ ] Update `php-stubs/wordpress-stubs` (via `composer update`)
* [ ] Update `sniccowp/php-scoper-wordpress-excludes`

**Notable things to test**

* https://github.com/WordPress/wordpress-develop/commit/7f7d616d822b79c952cbd6a3046a6f6a8aa5a35e
* REST API `prepare_links()` changes
* `WP_Query` caching

## References

[WordPress 6.1 Release Schedule](https://make.wordpress.org/core/6-1/)

Notable preliminary dates:

<table>
<tr>
	<td><b>Date
	<td><b>Event
<tr>
	<td>20 September 2022	
	<td>Beta 1
<tr>
	<td>11 October 2022	
	<td>Release Candidate 1
<tr>
	<td>1 November 2022
	<td>Stable Release
</table>

**Comment by spacedmonkey** [comment]
Two tickets I flagged in the meeting. 

1. [REST API: Avoid unnecessarily preparing item links](https://core.trac.wordpress.org/ticket/52992). 
TLDR, only call prepare_links method if needed. Example. 

```
   $response = rest_ensure_response( $data );
   if ( rest_is_field_included( '_links', $fields ) || rest_is_field_included( '_embedded', $fields ) ) {
	  $links = $this->prepare_links( $post );
	  $response->add_links( $links );
   }
```
2. [Adding caching to database queries in `WP_Query` ](https://core.trac.wordpress.org/ticket/22176)
WP_Query is now cached by default. This should help with performance, But may want to validate, we not need uncached queries in any context. 
4. [Use metadata api in *_network_options](https://core.trac.wordpress.org/ticket/37181). 
How network options are loaded and saved. Should have no check, but worth checking.

**Comment by spacedmonkey** [comment]
https://make.wordpress.org/core/2022/10/12/block-api-changes-in-wordpress-6-1/
